examples/thickWrapper.py

```diff
@@ -0,0 +1,245 @@
+import pyrevolve as pr
+import numpy as np
+
+# questions and todos
+#  - reverse operator is currently always called for a single step
+#  - if multiple time steps are needed to resume forward sweep, they need to be packed
+#    into the single 1D vector by the operator, and the Checkpointer object would not be
+#    aware of overlapping data if steps are close together.
+#  - only offline single-stage is supported at the moment
+#  - give user a good handle on verbosity
+
+#access descriptor
+# some magic (AD??) to determine which variables need to be recorded in the forward sweep
+# active: variable depends on independents and influences dependents in differentiable way
+#         do we need this at all as long as the user defines the adjoint operator?
+# in: input
+# out: output
+# TBR: as a first approximation: in && out
+# if in/out: needs to be stored in each iteration
+# if in: needs to be stored at start
+# if out: needs to be stored in fwd_rslt at end
+# if active: needs
+# in AD, TBR (to-be-recorded) is a more precise concept and removes variables that have no
+#    non-linear influence on the result, and hence no influence on the adjoint. This
+#    requires an analysis of the forward operator, and often a custom forward operator that
+#    computes only those variables that are actually needed for the reverse sweep. Maybe
+#    we'll have that in the future.
+
+class Checkpoint(object):
+    data = None
```

Comment:
This will create a class variable that is shared between different instances, which can created unwanted aliasing effects.